posts/heatmaps-clustering/index.qmd

```diff
@@ -0,0 +1,151 @@
+---
+title: "Heatmaps done right"
+subtitle: "A note on clustering distance matrices before plotting. Think sns.clustermap using altair."
+date: "08/03/2022"
+image: "visualization.png"
+categories: [visualisation, how-to]
+jupyter:
+  jupytext:
+    text_representation:
+      extension: .qmd
+      format_name: quarto
+      format_version: '1.0'
+      jupytext_version: 1.13.7
+  kernelspec:
+    display_name: Python 3.8.12 ('heatmap')
+    language: python
+    name: python3
+---
+
+Given a distance matrix, i.e. a symmetric matrix of distances between observations, if the indices of the observations are arbitrary, or more generally there is no variable by which we want to order the observations, then plotting a heatmap using the default ordering (of the indices) is often not very useful. A better approach is to cluster the observations, and use the new ordering. This is what `sns.clustermap` offers, see [docs](https://seaborn.pydata.org/generated/seaborn.clustermap.html).
+
+::: {.column-margin}
+_Question:_ Is there another/better way to re-order the indices? I don't think anything purely algebraic would work, as the before and after objects are fundamentally different matrices.
+:::
```

Comment:
@gbben this is mainly for you/us. could turn into a margin note (rather than question), or remove altogether.

Reply:
So what I understand here is that because you don't have a clear association across instances of the data having a default ordering doesn't make sense. Clustering will help in that it will order by cluster assignment, i.e. all those in cluster 1, cluster 2 etc. Is that correct? If so I think maybe this could do with a little more explanation. Maybe also specification that we need hard clustering (wait, do we? 🤔 ). In any case I think adding this will make it more obvious what you are trying to show here and will be a clearer opening gambit to the issue that you ask in the margin. Maybe :)

Reply:
left the question, but cleaned up as you suggested. [8d0fe96](https://github.com/bs-ds/blog/pull/7/commits/8d0fe968a854ca4625e0d4d1f38e60b12a951c8a)